core/client/fs/src/main/java/alluxio/util/FileSystemOptions.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.util;
+
+import alluxio.client.ReadType;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.grpc.CheckConsistencyPOptions;
+import alluxio.grpc.CreateDirectoryPOptions;
+import alluxio.grpc.CreateFilePOptions;
+import alluxio.grpc.DeletePOptions;
+import alluxio.grpc.ExistsPOptions;
+import alluxio.grpc.FileSystemMasterCommonPOptions;
+import alluxio.grpc.FreePOptions;
+import alluxio.grpc.GetStatusPOptions;
+import alluxio.grpc.ListStatusPOptions;
+import alluxio.grpc.LoadDescendantPType;
+import alluxio.grpc.LoadMetadataPOptions;
+import alluxio.grpc.LoadMetadataPType;
+import alluxio.grpc.MountPOptions;
+import alluxio.grpc.OpenFilePOptions;
+import alluxio.grpc.RenamePOptions;
+import alluxio.grpc.ScheduleAsyncPersistencePOptions;
+import alluxio.grpc.SetAclPOptions;
+import alluxio.grpc.SetAttributePOptions;
+import alluxio.grpc.TtlAction;
+import alluxio.grpc.UnmountPOptions;
+import alluxio.grpc.WritePType;
+import alluxio.security.authorization.Mode;
+
+/**
+ * This class contains static methods which can be passed Alluxio configuration objects that
+ * will populate the gRPC options objects with the proper values based on the given configuration.
+ */
+public class FileSystemOptions {
+
+  /**
+   * @param conf Alluxio configuration
+   * @return options based on the configuration
+   */
+  public static CreateDirectoryPOptions createDirectoryDefaults(AlluxioConfiguration conf) {
```

Comment:
Did you consider returning builder instead? That will decrease the merging overhead at BaseFileSystem. 
Or it could be both. createDirectoryDefaults() vs createDirectoryDefaultsBuilder(), the former can call the latter.

Reply:
I considered it, but it would require another big change because all of the methods that call this will now need to use `build()` in a lot of places. I think the change would be for the better but I put it off because of the amount of work required.

If it's something you'd like to see I can implement it once I address the rest of the comments